diff -u a/git-subtree.sh b/git-subtree.sh
--- a/git-subtree.sh	2016-07-30 10:43:00.187264000 +0200
+++ b/git-subtree.sh	2016-07-30 10:48:56.205607000 +0200
@@ -613,7 +613,7 @@
 	# We can't restrict rev-list to only $dir here, because some of our
 	# parents have the $dir contents the root, and those won't match.
 	# (and rev-list --follow doesn't seem to solve this)
-	grl='git rev-list --topo-order --reverse --parents $revs $unrevs'
+	grl='git rev-list --topo-order --reverse --parents --first-parent $revs $unrevs'
 	revmax=$(eval "$grl" | wc -l)
 	revcount=0
 	createcount=0
