diff -u a/reposurgeon b/reposurgeon
--- a/reposurgeon	2016-07-03 12:05:13.000000000 +0200
+++ b/reposurgeon	2016-08-02 01:32:33.713527039 +0200
@@ -9972,7 +9972,10 @@
                         # subcount 0 does not reliably work as it should
                         if filtercmd.startswith('--regex'):
                             try:
-                                self.regex = re.compile(parts[1].encode('ascii'))
+                                pattern = parts[1]
+                                if str is not bytes:
+                                    pattern = pattern.encode(master_encoding)
+                                self.regex = re.compile(pattern)
                             except sre_constants.error as e:
                                 raise Recoverable("filter compilation error - %s" % e)
                             self.sub = lambda s: polystr(self.regex.sub(polybytes(parts[2]),
